Library/Homebrew/cask/lib/hbc/cli/search.rb

```diff
@@ -15,8 +19,18 @@ def self.extract_regexp(string)
       end
 
       def self.search_remote(query)
-        matches = GitHub.search_code(user: "caskroom", path: "Casks",
-                                     filename: query, extension: "rb")
+        matches = begin
+          GitHub.search_code(
+            user: "caskroom",
+            path: "Casks",
+            filename: query,
+            extension: "rb",
+          )
+        rescue GitHub::Error => e
+          onoe e
+          $stderr.puts e.backtrace
```

Comment:
One doubt here: probably outputting the backtrace is a bit too much? `onoe e` will output in this case 

```
Error: curl failed!
curl: (6) Could not resolve host: api.github.com
curl: (3) <url> malformed
```

And the backtrace is not helpful for the user:

```
/usr/local/Homebrew/Library/Homebrew/utils/github.rb:226:in `raise_api_error'
/usr/local/Homebrew/Library/Homebrew/utils/github.rb:184:in `open'
/usr/local/Homebrew/Library/Homebrew/utils/github.rb:284:in `search'
/usr/local/Homebrew/Library/Homebrew/utils/github.rb:239:in `search_code'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/search.rb:23:in `search_remote'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/search.rb:58:in `search'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/search.rb:8:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/abstract_command.rb:35:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:97:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:167:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:131:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:8:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:95:in `<main>'
```

----

Probably the error message could be improved too (in `Library/Homebrew/utils/github.rb`). Like matching that error output from `curl`, checking the network connection and then doing `onoe 'Check your network connection'`. What do you think?

Reply:
I removed the stacktrace in 2a3f83b8d4ceeca370a60bbcb494990a2bd20801 and changed error to warning (as it doesn't fail the whole search). Now it looks like this:

```
$ brew cask search "foo-bar"
Warning: Online search failed: curl failed!
curl: (6) Could not resolve host: api.github.com
curl: (3) <url> malformed

No Cask found for "foo-bar".
```

Online/Github/Remote search?